File: app/src/main/java/za/co/duartbreedt/androidconceptsreference/MainActivity.java
package za.co.duartbreedt.androidconceptsreference;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import za.co.duartbreedt.androidconceptsreference.databinding.ActivityMainBinding;
import za.co.duartbreedt.androidconceptsreference.pusheen.PusheenDetailsActivity;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Inflate layout
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
    }

    @Override
    protected void onStart() {
        super.onStart();

        binding.buttonPrimary.setOnClickListener(clickedView -> onPrimaryButtonClicked());

        setupMyCustomView();
    }

    private void setupMyCustomView() {
        binding.myCustomView.setPrimaryText("My Custom View");
        binding.myCustomView.setSecondaryText("Yay!");
    }

    private void onPrimaryButtonClicked() {
        startActivity(new Intent(this, PusheenDetailsActivity.class));
    }
}

File: app/src/main/java/za/co/duartbreedt/androidconceptsreference/MainActivity.kt
package za.co.duartbreedt.androidconceptsreference

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import za.co.duartbreedt.androidconceptsreference.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Log.i("BOOP", "onCreate")

        binding = ActivityMainBinding.inflate(layoutInflater)

        setContentView(binding.root)
    }

    override fun onStart() {
        super.onStart()

        Log.i("BOOP", "onStart")
    }

    override fun onResume() {
        super.onResume()

        Log.i("BOOP", "onResume")
    }

    override fun onPause() {
        super.onPause()

        Log.i("BOOP", "onPause")
    }

    override fun onStop() {
        super.onStop()

        Log.i("BOOP", "onStop")
    }

    override fun onDestroy() {
        super.onDestroy()

        Log.i("BOOP", "onDestroy")
    }
}